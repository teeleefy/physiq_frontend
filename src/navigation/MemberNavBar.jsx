import { useContext } from "react";
import {MemberContext} from "../auth/UserContext";
import "./styles/MemberNavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import { Button, Card, CardBody, CardTitle, CardText, ListGroup } from "reactstrap";

/** Navigation bar for individual family members. Shows up on member's page.
 *
 * 
 * Rendered by Member.jsx.
 */

function MemberNavBar() {
    const { currentMember } = useContext(MemberContext);

    function renderNavLinks(){
        if(currentMember){
            return(
                <NavItem >
                    <NavLink id="MemberNavBar-name" className={"MemberNavBar-link"} to={`/member/${currentMember.id}`} key={currentMember.id}>{currentMember.firstName}</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/allergies`}>allergies</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/diagnoses`}>diagnoses</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/doctors`}>doctors</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/goals`}>goals</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/insurance`}>insurance</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/meds`}>meds</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/symptoms`}>symptoms</NavLink>
                    <NavLink className={"MemberNavBar-link"} to={`/member/${currentMember.id}/visits`}>visits</NavLink>
                </NavItem> 
            )
        }
        return (
                <NavItem>
                    <NavLink to="/">Home</NavLink>
                </NavItem>
        )
    }
 

  return (
    <div>
        <Navbar id="MemberNavBar">
            <Nav >
                {renderNavLinks()} 
            </Nav>
        </Navbar>
    </div>
  );
}

export default MemberNavBar;
