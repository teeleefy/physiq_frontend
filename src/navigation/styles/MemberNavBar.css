#MemberNavBar {
    background-color: red;
    width: 100%;
    height: 4rem;
  }
  
  #MemberNavBar-name {
    font-size: 2.3rem;
    margin-left:5rem;
    padding-right:1rem;
  }
  
  .MemberNavBar-link {
    color: black;
    text-decoration: none;
    box-sizing: border-box;
  }
  