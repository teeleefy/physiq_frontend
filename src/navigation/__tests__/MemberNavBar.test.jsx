import { renderWithRouterAndMemberContext, screen } from '../../../__tests__/test-tools';
import MemberNavBar from '../MemberNavBar'


describe('MemberNavBar', () => {
  it('renders the MemberNavBar component', () => {
    renderWithRouterAndMemberContext(<MemberNavBar />)
    
    screen.debug(); 
  })

  it("matches snapshot", function() {
    const { asFragment } = renderWithRouterAndMemberContext(<MemberNavBar />);
    expect(asFragment()).toMatchSnapshot();
  });

})
