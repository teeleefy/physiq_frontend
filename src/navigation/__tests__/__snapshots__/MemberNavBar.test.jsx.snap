// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MemberNavBar > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar"
      id="MemberNavBar"
    >
      <div
        class="container-fluid"
      >
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="MemberNavBar-link"
              href="/member/2"
              id="MemberNavBar-name"
            >
              Sarah
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/allergies"
            >
              allergies
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/diagnoses"
            >
              diagnoses
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/doctors"
            >
              doctors
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/goals"
            >
              goals
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/insurance"
            >
              insurance
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/meds"
            >
              meds
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/symptoms"
            >
              symptoms
            </a>
            <a
              class="MemberNavBar-link"
              href="/member/2/visits"
            >
              visits
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
